### PR TITLE
gzip content negotiation

### PR DESCRIPTION
**Pull request by dwerner**
This PR brings in content-negotiation using warp's `compression` feature. This should be considered an interim fix until something like https://github.com/seanmonstar/warp/pull/513 lands.

### CODE REVIEW

File: node/src/components/rpc_server/http_server.rs [hunk 1]
```diff
@@ -90,27 +90,35 @@ pub(super) async fn run<REv: ReactorEventT>(
...
+        .or(rpc_get_dictionary_item)
+        .or(rpc_query_global_state)
+        .or(unknown_method)
+        .or(parse_failure);
+
+    // Supports content negotiation for gzip responses. This is an interim fix until
+    // https://github.com/seanmonstar/warp/pull/513 moves forward.
+    let service_routes_gzip = warp::header::exact("accept-encoding", "gzip")
+        .and(service_routes.clone())
+        .with(warp::compression::gzip());
```
Comment (goral09) [inline node/src/components/rpc_server/http_server.rs:115]: Is this adding the `Content-Encoding` header automatically?

Comment (dwerner) [reply node/src/components/rpc_server/http_server.rs:115]: Yes, `warp`'s `compression` adds this to the response. Warp doesn't yet support automatically inspecting the `accept-encoding` header, so this does the check manually.

File: node/src/components/rpc_server/http_server.rs [hunk 2]
```diff
@@ -90,27 +90,35 @@ pub(super) async fn run<REv: ReactorEventT>(
...
-            .or(rpc_get_peers)
-            .or(rpc_get_status)
-            .or(rpc_get_era_info)
-            .or(rpc_get_auction_info)
-            .or(rpc_get_account_info)
-            .or(rpc_get_rpcs)
-            .or(rpc_get_dictionary_item)
-            .or(unknown_method)
-            .or(parse_failure),
+        service_routes_gzip.or(service_routes)
```
Comment (goral09) [inline node/src/components/rpc_server/http_server.rs]: Did you verify that we can have two set of the exact same endpoints that differ only in that new header? I.e. that we still can handle the requests without the `Accept-Encoding: gzip` header.

Comment (dwerner) [reply node/src/components/rpc_server/http_server.rs]: They are the same endpoints, they all just run this filter to inspect the accept-encoding. I manually tested that this works to support both variations using my new `read_trie` endpoint and tool (with gzip enabled and with it disabled, via the client-provided header).

**Review by goral09 — APPROVED** [review]
LGTM but please wait with the final approval for Fraser.

**Review by xcthulhu — APPROVED** [review]

**Review by Fraser999 — APPROVED** [review]

**Comment by dwerner** [comment]
Test plan for this is to hit any JSON-RPC endpoint with 
`accept-encoding: gzip` header set, as well as not set. I'd recommend using `curl --compressed` when set and just `curl` to verify that both gzip and plaintext responses are created.

**Pull request merged**